refactor(starknet_api): moved ResourceAsFelts block to starknet_api/transaction/fields.rs

ron-starkware opened the pull request.

reviewable-StarkWare commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/starkware-libs/sequencer/5862)
<!-- Reviewable:end -->

nimrod-starkware reviewed (changes requested):
> Reviewed 5 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/5862)* status: 5 of 6 files reviewed, 2 unresolved discussions (waiting on @barak-b-starkware)

___
*[`crates/starknet_api/src/lib.rs` line 53 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtJ8nQ22j34GvVqaVK:-ONtJ8nQ22j34GvVqaVL:undefined) ([raw file](https://github.com/starkware-libs/sequencer/blob/3254bfd26f20691dc418d6f7be40eb9e7fbf7820/crates/starknet_api/src/lib.rs#L53)):*
> ```XML
>     ParseIntError(#[from] ParseIntError),
>     #[error("Failed to convert from hex to Felt: {0:?} ")]
>     ResourceHexToFeltConversion(String),
> ```





_Suggestion:_
```XML
    #[error("Failed to convert the resource hex {0:?} to a felt.")]
    ResourceHexToFeltConversion(String),
```
___
*[`crates/starknet_api/src/transaction/fields.rs` line 297 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtItJC9lMXiCMg3O2s:-ONtItJC9lMXiCMg3O2t:b-wqhkyq) ([raw file](https://github.com/starkware-libs/sequencer/blob/3254bfd26f20691dc418d6f7be40eb9e7fbf7820/crates/starknet_api/src/transaction/fields.rs#L297)):*
> ```XML
>             max_amount: Felt::from(resource_bounds.max_amount),
>             max_price_per_unit: Felt::from(resource_bounds.max_price_per_unit),
>         })
> ```

please extract to a variable.



_Suggestion:_
```XML
        let resource_as_hex = resource.to_hex();
        Ok(Self {
            resource_name: Felt::from_hex(resource_as_hex).map_err(|_| {
                StarknetApiError::ResourceHexToFeltConversion(resource_as_hex.to_string())
            })?,
            max_amount: Felt::from(resource_bounds.max_amount),
            max_price_per_unit: Felt::from(resource_bounds.max_price_per_unit),
        })
```


<!-- Sent from Reviewable.io -->

ron-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/5862)* status: 4 of 6 files reviewed, 2 unresolved discussions (waiting on @barak-b-starkware and @nimrod-starkware)

___
*[`crates/starknet_api/src/transaction/fields.rs` line 297 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtItJC9lMXiCMg3O2s:-ONtLk3D8gHtm28RrrKA:b-g1gtr5) ([raw file](https://github.com/starkware-libs/sequencer/blob/3254bfd26f20691dc418d6f7be40eb9e7fbf7820/crates/starknet_api/src/transaction/fields.rs#L297)):*
<details><summary><i>Previously, nimrod-starkware wrote…</i></summary><blockquote>

please extract to a variable.
</blockquote></details>

Added a variable resource_as_hex
___
*[`crates/starknet_api/src/lib.rs` line 53 at r1](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtJ8nQ22j34GvVqaVK:-ONtLbm_-2Vw0mhTxHcD:bjl69ja) ([raw file](https://github.com/starkware-libs/sequencer/blob/3254bfd26f20691dc418d6f7be40eb9e7fbf7820/crates/starknet_api/src/lib.rs#L53)):*
> ```XML
>     ParseIntError(#[from] ParseIntError),
>     #[error("Failed to convert from hex to Felt: {0:?} ")]
>     ResourceHexToFeltConversion(String),
> ```

Fixed from "from hex to Felt: {0:?} " to "the resource hex {0:?} to a felt."


<!-- Sent from Reviewable.io -->

nimrod-starkware reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-:-ONtR91T3_r2SsbjwelR:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @barak-b-starkware)



<!-- Sent from Reviewable.io -->

barak-b-starkware commented:
*[`crates/starknet_api/src/transaction/fields.rs` line 290 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtYlt084Hnf8zBrdyX:-ONtYlt1BwUuYU9VZnH1:b-r96qtp) ([raw file](https://github.com/starkware-libs/sequencer/blob/ee84d8d487dbdac6773d5592e2c3bfaf6007895b/crates/starknet_api/src/transaction/fields.rs#L290)):*
> ```XML
>         resource: Resource,
>         resource_bounds: &ResourceBounds,
>     ) -> Result<Self, StarknetApiError> {
> ```

Does it work for you?



_Suggestion:_
```XML
    pub fn new(resource: Resource, resource_bounds: &ResourceBounds) -> StarknetApiResult<Self> {
```


<!-- Sent from Reviewable.io -->

nimrod-starkware reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/5862)* status: all files reviewed, 2 unresolved discussions (waiting on @barak-b-starkware and @ron-starkware)

___
*[`crates/starknet_os/src/vm_utils.rs` line 35 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtbI844EXjLkcD-2wR:-ONtmVCB4CBe_euOJ82L:b-mh5ktn) ([raw file](https://github.com/starkware-libs/sequencer/blob/ee84d8d487dbdac6773d5592e2c3bfaf6007895b/crates/starknet_os/src/vm_utils.rs#L35)):*
<details><summary><i>Previously, barak-b-starkware wrote…</i></summary><blockquote>

Isn't it weird not to use "transparent" here? @nimrod-starkware
</blockquote></details>

Does this imply that there is an implicit conversion from `StarknetApiError` to `VmUtilsError`?
If so, I would change it to `error(transparent)`.
But I don't think it implies that


<!-- Sent from Reviewable.io -->

ron-starkware reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/5862)* status: 4 of 6 files reviewed, 2 unresolved discussions (waiting on @barak-b-starkware and @nimrod-starkware)

___
*[`crates/starknet_api/src/transaction/fields.rs` line 290 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtYlt084Hnf8zBrdyX:-ONxhNDE4FLjiRLqVgap:b-886gsx) ([raw file](https://github.com/starkware-libs/sequencer/blob/ee84d8d487dbdac6773d5592e2c3bfaf6007895b/crates/starknet_api/src/transaction/fields.rs#L290)):*
<details><summary><i>Previously, barak-b-starkware wrote…</i></summary><blockquote>

Does it work for you?
</blockquote></details>

Fixed
___
*[`crates/starknet_os/src/vm_utils.rs` line 35 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtbI844EXjLkcD-2wR:-ONxhPRJ8wy780bJmUOZ:b-9n0prp) ([raw file](https://github.com/starkware-libs/sequencer/blob/ee84d8d487dbdac6773d5592e2c3bfaf6007895b/crates/starknet_os/src/vm_utils.rs#L35)):*
<details><summary><i>Previously, nimrod-starkware wrote…</i></summary><blockquote>

Does this imply that there is an implicit conversion from `StarknetApiError` to `VmUtilsError`?
If so, I would change it to `error(transparent)`.
But I don't think it implies that
</blockquote></details>

Changed to error(transparent)


<!-- Sent from Reviewable.io -->

nimrod-starkware reviewed (commented):
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/5862)* status: 5 of 6 files reviewed, 2 unresolved discussions (waiting on @barak-b-starkware)



<!-- Sent from Reviewable.io -->

ron-starkware commented:
*[`crates/starknet_os/src/vm_utils.rs` line 35 at r2](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-ONtbI844EXjLkcD-2wR:-ONyAmSY4QSBHD5T9d_W:b-m4ocyp) ([raw file](https://github.com/starkware-libs/sequencer/blob/ee84d8d487dbdac6773d5592e2c3bfaf6007895b/crates/starknet_os/src/vm_utils.rs#L35)):*
<details><summary><i>Previously, ron-starkware wrote…</i></summary><blockquote>

Changed to error(transparent)
</blockquote></details>

Reverted back


<!-- Sent from Reviewable.io -->

barak-b-starkware reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/sequencer/5862#-:-ONyB5HwElwR-_UIg4di:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ron-starkware)



<!-- Sent from Reviewable.io -->
